cmd/edit/cluster/cmd.go

```diff
@@ -473,6 +484,99 @@ func run(cmd *cobra.Command, _ []string) {
+			os.Exit(1)
+		}
+
+		auditLogRoleValue = args.auditLogRoleARN
+		auditLogRole = &auditLogRoleValue
+	}
+	if aws.IsHostedCP(cluster) {
+		// user did not supply value via flag in cli
+		if interactive.Enabled() && !auditLogRoleExists {
+			requestAuditLogForwarding, err := interactive.GetBool(interactive.Input{
+				Question: auditLogEnableOrUpdatePrompt(cluster.AWS().AuditLog().RoleArn()),
+				Default:  false,
+				Required: true,
+			})
+			if err != nil {
+				r.Reporter.Errorf("Expected a valid value: %s", err)
+				os.Exit(1)
+			}
+			if requestAuditLogForwarding {
+
+				r.Reporter.Infof("To configure the audit log forwarding role in your AWS account, " +
+					"please refer to steps 1 through 6: https://access.redhat.com/solutions/7002219")
+
+				auditLogRoleValue, err = interactive.GetString(interactive.Input{
+					Question: "Audit log forwarding role ARN",
+					Help:     cmd.Flags().Lookup("audit-log-arn").Usage,
+					Default:  auditLogRoleValue,
+					Required: true,
+					Validators: []interactive.Validator{
+						interactive.RegExp(aws.AuditLogArnRE.String()),
+					},
+				})
+				if err != nil {
+					r.Reporter.Errorf("Expected a valid value for audit-log-arn: %s", err)
+					os.Exit(1)
+				}
+				auditLogRole = &auditLogRoleValue
+			}
+			// user did not supply value via flag in cli
+		} else if interactive.Enabled() && auditLogRoleExists {
+			disableAuditLog, err := interactive.GetBool(interactive.Input{
+				Question: "Disable Audit Log",
+				Help:     "Disable AWS CloudWatch audit log forwarding that is currently enabled.",
+				Default:  false,
+			})
+			if err != nil {
+				r.Reporter.Errorf("Expected a valid value: %s", err)
+				os.Exit(1)
+			}
+			if disableAuditLog {
+				auditLogRoleValue = ""
+				auditLogRole = &auditLogRoleValue
+			}
+
+			// validation for user supplied value via flag
+		} else if auditLogRoleValue != "" && !auditLogRoleExists {
+			if !aws.AuditLogArnRE.MatchString(auditLogRoleValue) {
+				r.Reporter.Errorf("Expected a valid value for audit-log-arn matching %s", aws.AuditLogArnRE.String())
+				os.Exit(1)
+			}
+			r.Reporter.Warnf("You are enabling audit log forwarding with the provided arn '%s'.", auditLogRoleValue)
+			if !confirm.Confirm("enable audit log forwarding for cluster '%s'", clusterKey) {
+				os.Exit(0)
+			}
+			// validation for user supplied value via flag
+		} else if auditLogRoleValue != "" && auditLogRoleExists {
+			if !aws.AuditLogArnRE.MatchString(auditLogRoleValue) {
+				r.Reporter.Errorf("Expected a valid value for audit-log-arn matching %s", aws.AuditLogArnRE.String())
+				os.Exit(1)
+			}
+			r.Reporter.Warnf("You are editing audit log forwarding with the provided arn '%s'.", auditLogRoleValue)
+			if !confirm.Confirm("edit audit log forwarding for cluster '%s'", clusterKey) {
+				os.Exit(0)
+			}
+			// validation for user supplied value via flag
+		} else if auditLogRoleValue == "" && auditLogRoleExists {
```

Comment:
I don't think we do anything like that for commands today. So because of that I'd say it is overkill. But if this is something that has been worked out with BU then we move it forward

Reply:
I don't believe there was any disagreement from BU side, when I added the screen shots of the reactive prompts (within the Jira Card associated with this PR)

Hi @ciaranRoche @oriAdler 
would like to get consensus / vote on what would be the best approach here to better the user experience as well as meeting the standards for the repo. Please kindly advise.  :slightly_smiling_face:

Reply:
If we can somehow break it into smaller functions, or rearrange the conditions, maybe a `switch` statement? I don't know.
That could be great.
I have no doubt the user experience here is good, but I agree with Guilherme there are a lot of `else if` conditions here.

Reply:
Yeah i agree with both Ori and Guilherme here. If we can flatten this out, potentially abstract to smaller functions and have unit tests on it to make sure we do not break it in the future.

Reply:
Thank you all for the review:
I went ahead and refactored the code to reduce multiple if else if statements.

Reply:
![image](https://github.com/openshift/rosa/assets/118839428/1a538d6c-efc1-4037-9c38-6854c2db6285)
First statement covers case when user explicitly defines the flag / arg value for audit log role arn
The if statement following, covers case when user did not define a value and must go through the interactive mode